src/client/mod.rs

```diff
@@ -32,6 +33,12 @@ pub trait KakarotTransactions {
     async fn send_raw_transaction(&self, transaction: Bytes) -> EthApiResult<B256>;
 }
 
+#[async_trait]
+pub trait TransactionHashProvider {
+    /// Returns the transaction by hash.
+    async fn transaction_by_hash(&self, hash: B256) -> EthApiResult<Option<WithOtherFields<Transaction>>>;
+}
+
```

Comment:
can't this do to the `TransactionProvider` trait?

Reply:
Not exactly because I have to upstream only this part of the trait to be able to check the `mempool` at `eth_client` level (because the mempool is not available at the provider level).

So what I do here is that:
- I implement this overall function for eth client.
- It allows me to check if I find my transaction in the mempool.
     - If it is the case I return this transaction which is therefore pending.
     - If it is not the case, I call the subfunction `transaction_by_hash` which is in the provider (the old function) but which will only look in the database (the finalized transactions) like that I have not touched this part.

Reply:
ahh ok missed that part sorry